src/Plugins/Nop.Plugin.Seed.ProductSync/Services/ProductSyncService.cs

```diff
@@ -0,0 +1,342 @@
+                                    valueEntity.WeightAdjustment = value.WeightAdjustment;
+                                    valueEntity.AttributeValueType = AttributeValueType.Simple;
+
+                                    await _productAttributeService.UpdateProductAttributeValueAsync(valueEntity);
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        var productAttributeMapping = new ProductAttributeMapping()
+                        {
+                            AttributeControlTypeId = attribute.AttributeControlType,
+                            IsRequired = attribute.IsRequired,
+                            ProductId = productId,
+                            ProductAttributeId = productAttribute.Id
+                        };
+                        await _productAttributeService.InsertProductAttributeMappingAsync(productAttributeMapping);
+                        foreach (var value in attribute.ProductAttributeValues)
+                        {
+                            var productAttributeValue = new ProductAttributeValue()
+                            {
+                                ProductAttributeMappingId = productAttributeMapping.Id,
+                                Name = value.Name,
+                                PriceAdjustment = value.PriceAdjustment,
+                                IsPreSelected = value.IsPreSelected,
+                                DisplayOrder = 1,
+                                AttributeValueType = AttributeValueType.Simple,
+                                WeightAdjustment = value.WeightAdjustment,
+                            };
+                            await _productAttributeService.InsertProductAttributeValueAsync(productAttributeValue);
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    public async Task<ProductPicture> GetProductPicture(string thumbnail, int productId)
+    {
+        var fileExtention = Path.GetExtension(Path.GetFileName(thumbnail))?.Remove(0,1);
+        var mimeType = $"image/{fileExtention}";
+        var byteData = await GetByteDataFromUrl(thumbnail);
+        var filename = Path.GetFileNameWithoutExtension(thumbnail);
+        var picture = await _pictureService.InsertPictureAsync(byteData, mimeType, filename);
+        return new ProductPicture() { PictureId = picture.Id, ProductId = productId };
+    }
+
+    public async Task<Product> GetProductByInfigoId(int id)
+    {
+        var tagName = $"infigo_product_{id}";
+        var tags = await _productTagService.GetAllProductTagsAsync(tagName);
+        if (!tags.IsNullOrEmpty())
+        {
+            var first = tags?.First();
+            var productProductTagMappings = await _productProductTagMapping.Table.FirstOrDefaultAsync(x=>x.ProductTagId==first.Id);
```

Comment:
instead of doing this here, and then again in line 326, just create a method in ProductService.
GetByTag("infigo_product_{31}");, and then return product in 1 go